server/src/main/java/io/crate/execution/engine/aggregation/impl/average/numeric/NumericAverageState.java

```diff
@@ -44,7 +45,8 @@ public class NumericAverageState<T extends NumericValueHolder> implements Compar
 
     public BigDecimal value() {
         if (count > 0) {
-            return sum.value().divide(BigDecimal.valueOf(count));
+            // Divide uses same rounding mode as regular division does.
+            return sum.value().divide(BigDecimal.valueOf(count), MathContext.DECIMAL64);
```

Comment:
Proper fix would be using user defined returnType in the NumericAverageState

Reply:
I was wrong, we already use stored type as return type and use it in iterate to cast Object args. 

For average itself we want to use best precision (same with, see https://github.com/crate/crate/issues/17357#issuecomment-2643075627). If users want to cut precision, Cratedb respects explicit casts, see https://github.com/crate/crate/issues/17357#issuecomment-2643419489